src/components/Cell/Cell.tsx

```diff
@@ -35,26 +41,94 @@ const Cell = ({ id, bullDoze, setBullDoze, filterGarden, setFilterGarden, toggle
     setApiError(error)
   }
 
+  useEffect(() => {
+    if (garden) {
+      const foundCell = garden.find(cell => cell.id === id)
+      if (foundCell !== undefined && foundCell.status === 1) {
+        setIsDisabled(true);
+        !isDisabled ? setClassName('cell disabled') : setClassName('cell');
+      }
+      if (foundCell !== undefined && foundCell.plant_id) {
+        setCellContents({ plant: foundCell })
+        setIsPopulated(true);
+        setIsPlanted(true);
+      }
+      setCell(foundCell);
+    }
+    // eslint-disable-next-line
+  }, []);
+
+  useEffect(() => {
+    setGarden((prevState: GardenKeys) => {
+      let index = prevState?.findIndex((item) => item.id === cell?.id);
+      let newState = [...prevState];
+      newState[index] = {
+        ...newState[index],
+        status: Number(isDisabled)
+      };
+      return newState;
+    });
+    // eslint-disable-next-line
+  }, [isDisabled]);
```

Comment:
We discussed - make sure to refactor this to only update the single cell data instead of updating the entire garden state